include/fmt/color.h

```diff
@@ -563,6 +563,52 @@ inline std::basic_string<Char> format(const text_style& ts, const S& format_str,
                  fmt::make_args_checked<Args...>(format_str, args...));
 }
 
+/** Formats a string with the given text_style and writes the output to ``out``.
+ */
+// GCC 8 and earlier cannot handle std::back_insert_iterator<Container> with
+// vformat_to<ArgFormatter>(...) overload, so SFINAE on iterator type instead.
+template <typename OutputIt, typename S, typename Char = char_t<S>,
+          FMT_ENABLE_IF(detail::is_output_iterator<OutputIt>::value)>
+OutputIt vformat_to(
+    OutputIt out, const text_style& ts, const S& format_str,
+    basic_format_args<buffer_context<type_identity_t<Char>>> args) {
+  decltype(detail::get_buffer<Char>(out)) buf(detail::get_buffer_init(out));
+  detail::vformat_to(buf, ts, to_string_view(format_str), args);
+  return detail::get_iterator(buf);
+}
+
+/**
+ \rst
+ Formats arguments with the given text_style, writes the result to the output
+ iterator ``out`` and returns the iterator past the end of the output range.
+
+ **Example**::
+
+   std::vector<char> out;
+   fmt::format_to(std::back_inserter(out),
+                  fmt::emphasis::bold | fg(fmt::color::red), "{}", 42); \endrst
+ */
+template <typename OutputIt, typename S, typename... Args,
+          FMT_ENABLE_IF(detail::is_output_iterator<OutputIt>::value&&
+                            detail::is_string<S>::value)>
+inline OutputIt format_to(OutputIt out, const text_style& ts, const S& format_str,
+                          Args&&... args) {
+  basic_format_args<buffer_context<type_identity_t<char_t<S>>>> store =
+      fmt::make_args_checked<Args...>(format_str, args...);
+  return vformat_to(out, ts, to_string_view(format_str), store);
+}
+
+template <typename S, typename... Args, size_t SIZE = inline_buffer_size,
+          typename Char = enable_if_t<detail::is_string<S>::value, char_t<S>>>
+inline typename buffer_context<Char>::iterator format_to(
+    basic_memory_buffer<Char, SIZE>& buf, const text_style& ts,
+    const S& format_str, Args&&... args) {
+  basic_format_args<buffer_context<type_identity_t<Char>>> store =
+      fmt::make_args_checked<Args...>(format_str, args...);
+  detail::vformat_to(buf, ts, to_string_view(format_str), store);
+  return detail::buffer_appender<Char>(buf);
+}
```

Comment:
I don't think this is necessary.

Reply:
Thank you for your feedback. I corrected my PR accordingly.
I really would like to keep this overload though, mainly it is the main reason why I submitted this PR.

This overload integrates well into a `struct fmt::formatter<` specializations for custom types,
it is also possible to colorize only parts of a string and for multiple colorized parts a memory allocation adds up fast.

Otherwise we would have to use the output iterator overload for writing into a `memory_buffer` although the private color API accepts a `memory_buffer` already but is inaccessible. This would add unneccessary code overhead (and probably adds a performance overhead too).

Reply:
This overload is redundant because it's equivalent to passing back_insert_iterator to memory_buffer. There should be no overhead.